src/gpu/vulkan/SDL_gpu_vulkan.c

```diff
@@ -11285,6 +11286,24 @@ static Uint8 VULKAN_INTERNAL_IsDeviceSuitable(
         return 0;
     }
 
+    /* If we prefer high performance, sum up all device local memory (rounded to megabytes)
+     * to deviceRank. In the niche case of someone having multiple dedicated GPUs in the same
+     * system, this theoretically picks the most powerful one (or at least the one with the
+     * most memory!)
+     */
+
+    renderer->vkGetPhysicalDeviceMemoryProperties(physicalDevice, &deviceMemory);
+    Uint64 videoMemory = 0;
+    for (i = 0; i < deviceMemory.memoryHeapCount; i++) {
+        VkMemoryHeap heap = deviceMemory.memoryHeaps[i];
+        if (heap.flags & VK_MEMORY_HEAP_DEVICE_LOCAL_BIT) {
+            videoMemory += heap.size;
+        }
+    }
+    // Round it to megabytes (as per the vulkan spec videoMemory is in bytes)
+    Uint64 videoMemoryRounded = videoMemory / 1024 / 1024;
+    *deviceRank += videoMemoryRounded;
```

Comment:
Will adding a memory number directly to `deviceRank` effectively discard the GPU type preference specified by `DEVICE_PRIORITY_HIGHPERFORMANCE` and `DEVICE_PRIORITY_LOWPOWER`, most of the time?

Reply:
I don't think so, since there are several device types.